app/gui/src/log.ts

```diff
@@ -0,0 +1,85 @@
+/** @file Application-wide logging setup by overriding standard console methods. */
+
+/** Helper function for displaying errors in logs. */
+function serializeError(err: unknown) {
+  if (err instanceof Error) {
+    return {
+      name: err.name,
+      message: err.message,
+      stack: err.stack,
+      cause: err.cause ? String(err.cause) : undefined,
+    }
+  }
+  try {
+    return { message: JSON.stringify(err) }
+  } catch {
+    return { message: String(err) }
+  }
+}
+
+/**
+ * Override standard console methods with our own, that also log to the file using Electron.
+ * Also setup event listeners for unhandled errors and resource load errors.
+ *
+ * Only the following methods are overridden:
+ * - `console.log`
+ * - `console.error`
+ * - `console.warn`
+ * - `console.info`
+ */
+export function setupLogger() {
+  const originalConsoleLog = window.console.log
+  const originalConsoleError = window.console.error
+  const originalConsoleWarn = window.console.warn
+
+  window.console.log = (...args) => {
+    originalConsoleLog.apply(window.console, args)
+    window.logApi?.log(args)
+  }
+  window.console.info = (...args) => {
+    window.console.log(...args)
+  }
```

Comment:
I got confused by the fact that:
* The info in electron client is _not_ logged in the file
* But in web app it _is_ because it becomes `log` ;evel here.

Either write info to the file or _not_ convert to log here and mention that in this function's docs.